core/src/main/java/io/confluent/kafka/schemaregistry/tools/CheckSchemaCompatibility.java

```diff
@@ -0,0 +1,395 @@
+      List<String> sortedOnlyInTarget = new ArrayList<>(onlyInTarget);
+      Collections.sort(sortedOnlyInTarget);
+      for (String subject : sortedOnlyInTarget) {
+        LOG.info("  - {}", subject);
+      }
+    }
+
+    if (sourceSet.isEmpty() || targetSet.isEmpty()) {
+      LOG.info("\n✓ Compatible, one schema registry is empty");
+      return true;
+    }
+
+    // Check if target has subjects that don't exist in source (not acceptable)
+    if (!onlyInTarget.isEmpty()) {
+      LOG.error("\n✗ Not compatible, target has subjects that don't exist in source: {}",
+          onlyInTarget);
+      return false;
+    }
+
+    // If source has more subjects than target, that's acceptable (informational only)
+    if (!onlyInSource.isEmpty()) {
+      LOG.info("\nℹ Source has additional subjects (acceptable): {}", onlyInSource);
+    }
+
+
+    if (!inBoth.isEmpty()) {
+      LOG.info("\nSubjects in both registries:");
+      List<String> sortedInBoth = new ArrayList<>(inBoth);
+      Collections.sort(sortedInBoth);
+      for (String subject : sortedInBoth) {
+        if (!compareSubject(sourceClient, targetClient, subject)) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  private boolean compareSubject(SchemaRegistryClient sourceClient, 
+                                 SchemaRegistryClient targetClient, 
+                                 String subject) {
+    try {
+      LOG.info("\n--- Comparing subject: {} ---", subject);
+      
+      // Get all versions for the subject from both registries
+      List<Integer> sourceVersions = sourceClient.getAllVersions(subject);
+      List<Integer> targetVersions = targetClient.getAllVersions(subject);
+      
+      LOG.info("Source versions: {}", sourceVersions);
+      LOG.info("Target versions: {}", targetVersions);
+
+      // Find versions that exist in both registries
+      Set<Integer> sourceVersionSet = new HashSet<>(sourceVersions);
+      Set<Integer> targetVersionSet = new HashSet<>(targetVersions);
+      Set<Integer> commonVersions = new HashSet<>(sourceVersionSet);
+      commonVersions.retainAll(targetVersionSet);
+      
+      // Find versions only in source or target
+      Set<Integer> onlyInSource = new HashSet<>(sourceVersionSet);
+      onlyInSource.removeAll(targetVersionSet);
+      Set<Integer> onlyInTarget = new HashSet<>(targetVersionSet);
+      onlyInTarget.removeAll(sourceVersionSet);
+      
+      // Log version differences
+      if (!onlyInSource.isEmpty()) {
+        LOG.info("ℹ Versions only in source (will be ignored): {}", onlyInSource);
+      }
+      if (!onlyInTarget.isEmpty()) {
+        LOG.error("✗ Target has versions that don't exist in source for subject '{}': {}", 
+                 subject, onlyInTarget);
+        return false;
+      }
+      
+      if (commonVersions.isEmpty()) {
+        LOG.error("✗ No common versions found for subject '{}'", subject);
```

Comment:
Is this valid case? This will only happen if subject is empty? 
I don't think we should fail in this case.